ui/components/app/confirm-gas-display/confirm-legacy-gas-display/confirm-legacy-gas-display.js

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  checkNetworkAndAccountSupports1559,
+  getIsMainnet,
+  getPreferences,
+  getUnapprovedTransactions,
+  getUseCurrencyRateCheck,
+  transactionFeeSelector,
+  txDataSelector,
+} from '../../../../selectors';
+import { hexWEIToDecGWEI } from '../../../../../shared/modules/conversion.utils';
+import { isLegacyTransaction } from '../../../../helpers/utils/transactions.util';
+import { PRIMARY, SECONDARY } from '../../../../helpers/constants/common';
+
+import GasTiming from '../../gas-timing';
+import TransactionDetailItem from '../../transaction-detail-item';
+import UserPreferencedCurrencyDisplay from '../../user-preferenced-currency-display';
+import InfoTooltip from '../../../ui/info-tooltip';
+import LoadingHeartBeat from '../../../ui/loading-heartbeat';
+import { Text } from '../../../component-library/text';
+import {
+  FONT_STYLE,
+  TextVariant,
+  TextColor,
+} from '../../../../helpers/constants/design-system';
+
+const renderHeartBeatIfNotInTest = () =>
+  process.env.IN_TEST ? null : <LoadingHeartBeat />;
+
+const ConfirmLegacyGasDisplay = ({ draftTransaction = {} }) => {
+  const t = useI18nContext();
+
+  // state selectors
+  const isMainnet = useSelector(getIsMainnet);
+  const useCurrencyRateCheck = useSelector(getUseCurrencyRateCheck);
+  const { useNativeCurrencyAsPrimaryCurrency } = useSelector(getPreferences);
+  const unapprovedTxs = useSelector(getUnapprovedTransactions);
+  let transactionData = {};
+  if (Object.keys(draftTransaction).length !== 0) {
```

Comment:
Hey not sure can these be different 
- unapprovedTxs[draftTransaction.id]
- draftTransaction